### PR TITLE
Added ShoppingCartController to manage ShoppingCart and items in it. …

### PR DESCRIPTION
**Pull request by Zel1oy**
…Provided all stuff needed

### CODE REVIEW

**Review by IhorKalaur — APPROVED** [review]

File: src/main/java/com/example/onlinebookstore/config/SecurityConfig.java [hunk 1]
```diff
@@ -36,7 +38,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
         return http.cors(AbstractHttpConfigurer::disable)
                 .csrf(AbstractHttpConfigurer::disable)
                 .authorizeHttpRequests(
-                        auth -> auth.requestMatchers("/auth/**")
+                        auth -> auth.requestMatchers(AUTHORIZATION_URL,
+                                        "/swagger-ui/**", "/v3/api-docs/**", ERROR_URL)
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/config/SecurityConfig.java]: If you created constants for AUTHORIZATION_URL please make constants for other URLs to be consistent

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @Override
+    public ShoppingCartDto addToCart(CreateCartItemRequestDto request) {
+        User user = userService.getAuthenticatedUser();
+        Book book = bookRepository.findById(request.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Book not found"));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createNewShoppingCart(user));
+
+        CartItem cartItem = new CartItem();
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: Move logic of creation and population to separate method

File: src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        User user = userService.getAuthenticatedUser();
+        return shoppingCartMapper.toDto(shoppingCartRepository.findByUserId(user.getId())
+                .orElseGet(() -> createNewShoppingCart(user)));
+    }
+
+    @Override
+    public ShoppingCartDto addToCart(CreateCartItemRequestDto request) {
+        User user = userService.getAuthenticatedUser();
+        Book book = bookRepository.findById(request.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Book not found"));
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java]: add details to error message

File: src/main/resources/db/changelog/changes/10-create-cart-items-table.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+              - column:
+                  name: quantity
+                  type: integer
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: boolean
+                  constraints:
+                    nullable: false
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/10-create-cart-items-table.yaml]: Please add empty lines to the end of each file

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ, but some improvements required

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**